tests/test_loader.py

```diff
@@ -630,367 +227,6 @@ def test_avoid_reprocessing_without_initial_values_list(self):
-        self.assertEqual(proc(['', 'hello', 'world']), u' hello world')
-        self.assertEqual(proc(['hello', 'world']), u'hello world')
-        self.assertIsInstance(proc(['hello', 'world']), str)
-
-    def test_compose(self):
-        proc = Compose(lambda v: v[0], str.upper)
-        self.assertEqual(proc(['hello', 'world']), 'HELLO')
-        proc = Compose(str.upper)
-        self.assertEqual(proc(None), None)
-        proc = Compose(str.upper, stop_on_none=False)
-        self.assertRaises(ValueError, proc, None)
-        proc = Compose(str.upper, lambda x: x + 1)
-        self.assertRaises(ValueError, proc, 'hello')
-
-    def test_mapcompose(self):
-        def filter_world(x):
-            return None if x == 'world' else x
-        proc = MapCompose(filter_world, str.upper)
-        self.assertEqual(proc([u'hello', u'world', u'this', u'is', u'scrapy']),
-                         [u'HELLO', u'THIS', u'IS', u'SCRAPY'])
-        proc = MapCompose(filter_world, str.upper)
-        self.assertEqual(proc(None), [])
-        proc = MapCompose(filter_world, str.upper)
-        self.assertRaises(ValueError, proc, [1])
-        proc = MapCompose(filter_world, lambda x: x + 1)
-        self.assertRaises(ValueError, proc, 'hello')
-
-
-class SelectortemLoaderTest(unittest.TestCase):
-    response = HtmlResponse(url="", encoding='utf-8', body=b"""
-    <html>
-    <body>
-    <div id="id">marta</div>
-    <p>paragraph</p>
-    <a href="http://www.scrapy.org">homepage</a>
-    <img src="/images/logo.png" width="244" height="65" alt="Scrapy">
-    </body>
-    </html>
-    """)
-
-    def test_init_method(self):
-        l = TestItemLoader()
-        self.assertEqual(l.selector, None)
-
-    def test_init_method_errors(self):
-        l = TestItemLoader()
-        self.assertRaises(RuntimeError, l.add_xpath, 'url', '//a/@href')
-        self.assertRaises(RuntimeError, l.replace_xpath, 'url', '//a/@href')
-        self.assertRaises(RuntimeError, l.get_xpath, '//a/@href')
-        self.assertRaises(RuntimeError, l.add_css, 'name', '#name::text')
-        self.assertRaises(RuntimeError, l.replace_css, 'name', '#name::text')
-        self.assertRaises(RuntimeError, l.get_css, '#name::text')
-
-    def test_init_method_with_selector(self):
-        sel = Selector(text=u"<html><body><div>marta</div></body></html>")
-        l = TestItemLoader(selector=sel)
-        self.assertIs(l.selector, sel)
-
-        l.add_xpath('name', '//div/text()')
-        self.assertEqual(l.get_output_value('name'), [u'Marta'])
-
-    def test_init_method_with_selector_css(self):
-        sel = Selector(text=u"<html><body><div>marta</div></body></html>")
-        l = TestItemLoader(selector=sel)
-        self.assertIs(l.selector, sel)
-
-        l.add_css('name', 'div::text')
-        self.assertEqual(l.get_output_value('name'), [u'Marta'])
-
-    def test_init_method_with_response(self):
-        l = TestItemLoader(response=self.response)
```

Comment:
I think we should be keeping tests like that. This removal probably needs a more detailed review: everything which is related to response, or e.g. scrapy.Item, should be tested, because these are features which Scrapy's wrapper provides, not features which itemloaders library provides.

I also wonder if many (most) other tests needs to be preserved: e.g. I can see how we could have implemented a scrapy ItemLoader subclass in a way which breaks itemloaders.ItemLoader's `selector` argument. Though we haven't done this for scrapy.Selector, so no strong opinion here.

Reply:
Makes sense to me.
The ones I removed were related to a usage of `response` in which I transformed to `selector` in `itemloaders`.
But I'm in favor of keeping the ones referencing `response`..